core/app/service/backup.go

```diff
@@ -107,13 +107,14 @@ func (u *BackupService) GetLocalDir() (string, error) {
 }
 
 func (u *BackupService) SearchWithPage(req dto.SearchPageWithType) (int64, interface{}, error) {
-	count, accounts, err := backupRepo.Page(
-		req.Page,
-		req.PageSize,
-		repo.WithByType(req.Type),
-		repo.WithByName(req.Info),
-		repo.WithOrderBy("created_at desc"),
-	)
+	options := []global.DBOption{repo.WithOrderBy("created_at desc")}
+	if len(req.Type) != 0 {
+		options = append(options, repo.WithByType(req.Type))
+	}
+	if len(req.Info) != 0 {
+		options = append(options, repo.WithByType(req.Info))
+	}
+	count, accounts, err := backupRepo.Page(req.Page, req.PageSize, options...)
 	if err != nil {
 		return 0, nil, err
 	}
```

Comment:
There are no apparent inconsistencies between the original code snippet provided and the current version of it.

This is good practice to keep files up-to-date with the latest changes. It ensures that new features or fixes do not get lost in older versions when looking at different parts of the same codebase.